modules/sms_user/src/AccountRegistration.php

```diff
@@ -0,0 +1,363 @@
+   * @param \Drupal\sms\Provider\SmsProviderInterface $sms_provider
+   *   The SMS provider.
+   * @param \Drupal\sms\Provider\PhoneNumberProviderInterface $phone_number_provider
+   *   The phone number provider.
+   */
+  public function __construct(ConfigFactoryInterface $config_factory, Token $token, SmsProviderInterface $sms_provider, PhoneNumberProviderInterface $phone_number_provider) {
+    $this->configFactory = $config_factory;
+    $this->token = $token;
+    $this->smsProvider = $sms_provider;
+    $this->phoneNumberProvider = $phone_number_provider;
+
+    // @fixme, use phone number provider to get settings.
+    // @see https://www.drupal.org/node/2709465
+//      $phone_number_settings = $this->phoneNumberProvider->getPhoneNumberSettings('user', 'usxer');
+    $this->userPhoneNumberSettings = PhoneNumberSettings::load('user.user');
+
+    // @fixme.  number resolution should move to a method on
+    // @see https://www.drupal.org/node/2709463
+    $this->phoneNumberVerificationStorage = \Drupal::entityTypeManager()
+      ->getStorage('sms_phone_number_verification');
+
+    $this->settings = $this->configFactory
+      ->get('sms_user.settings')
+      ->get('account_registration');
+  }
+
+  /**
+   * @inheritdoc
+   */
+  public function createAccount(SmsMessageInterface $sms_message) {
+    $sender_number = $sms_message->getSenderNumber();
+    if (!empty($sender_number)) {
+      // Any users with this phone number?
+      $count = $this->phoneNumberVerificationStorage
+        ->getQuery()
+        ->condition('entity__target_type', 'user')
+        ->condition('phone', $sender_number)
+        ->count()
+        ->execute();
+
+      if (!$count) {
+        if (!empty($this->settings['all_unknown_numbers']['status'])) {
+          $this->allUnknownNumbers($sms_message);
+        }
+        if (!empty($this->settings['formatted']['status'])) {
+          $this->preFormattedMessage($sms_message);
+        }
+      }
+    }
+  }
+
+  /**
+   * Process an incoming message and create a user if the phone number is
+   * unrecognised.
+   *
+   * @param \Drupal\sms\Entity\SmsMessageInterface $sms_message
+   *   An incoming SMS message.
+   */
+  protected function allUnknownNumbers(SmsMessageInterface $sms_message) {
+    /** @var \Drupal\user\UserInterface $user */
+    $user = User::create(['name' => $this->generateUniqueUsername()]);
+    $user->activate();
+
+    // Sender phone number.
+    $sender_number = $sms_message->getSenderNumber();
+    $t_args['%sender_phone_number'] = $sender_number;
+    $phone_field_name = $this->userPhoneNumberSettings->getFieldName('phone_number');
+    $user->{$phone_field_name}[] = $sender_number;
+
+    // Password.
+    $password = user_password();
+    $user->setPassword($password);
+
+    $validate = $user->validate();
+    if ($validate->count() == 0) {
+      $user->save();
+
+      // @todo autoconfirm the number?
+      // @see https://www.drupal.org/node/2709911
```

Comment:
Considering that this is the process for confirmation of a number, we can assume the number confirmed.